timemachine/ff/handlers/openmm_deserializer.py

```diff
@@ -85,9 +91,17 @@ def deserialize_nonbonded_force(force, N):
         else:
             lj_scale_factor = 1 - new_eps / expected_eps
 
-        scale_factors_.append(lj_scale_factor)
+        if expected_q == 0:
+            if new_q == 0:
+                q_scale_factor = 1
+            else:
+                raise RuntimeError("Divide by zero in charge calculation")
+        else:
+            q_scale_factor = 1 - new_q / expected_q
```

Comment:
Don't 100% understand why a divide-by-zero is possible here. (But this works as-is: returns a solution to "(1 - scale_factor) * expected_q = new_q".)

Reply:
Roughly the statement is saying: if the *desired* charge product is non-zero, but we're *starting* with a zero charge product, there's no number that we can multiply a zero starting charge with to arrive at a non-zero desired charge. The naming is confusing - I will rename it.